packages/backend-core/src/users/utils.ts

```diff
@@ -16,30 +16,55 @@ export const hasAdminPermissions = sdk.users.hasAdminPermissions
 export const hasBuilderPermissions = sdk.users.hasBuilderPermissions
 export const hasAppBuilderPermissions = sdk.users.hasAppBuilderPermissions
 
-export async function isCreator(user?: User | ContextUser) {
+async function getGroups(groupIds: string[]) {
+  if (groupIds.length) {
+    const db = context.getGlobalDB()
+    return await db.getMultiple<UserGroup>(groupIds, { allowMissing: true })
+  }
+  return []
+}
+
+export async function creatorsInList(
+  users: (User | ContextUser)[],
+  groups?: UserGroup[]
+) {
+  if (!groups) {
+    const groupIds = users.flatMap(user => user.userGroups || [])
+    if (groupIds.length) {
```

Comment:
You don't need this check thanks to the optimisation you've done in `getGroups`. This then allows you to get rid of the `|| []` a few lines below.